### ISSUE TITLE
The `GlobalImportFilter` imports the Globals after the Macros and therefor "overwrites" the Macros:

### ISSUE THREAD

**Issue by pmario** [TiddlyWiki/TiddlyWiki5#8683]
>@pmario - the `GlobalImportFilter` imports the Globals after the Macros and therefor "overwrites" the Macros:

```
[[$:/core/ui/PageMacros]] [all[shadows+tiddlers]tag[$:/tags/Macro]!is[draft]] [all[shadows+tiddlers]tag[$:/tags/Global]!is[draft]]
```

I believe it should be this way around:

```
[[$:/core/ui/PageMacros]] [all[shadows+tiddlers]tag[$:/tags/Global]!is[draft]] [all[shadows+tiddlers]tag[$:/tags/Macro]!is[draft]]
```

_Originally posted by @BurningTreeC in https://github.com/TiddlyWiki/TiddlyWiki5/issues/8678#issuecomment-2416757479_

**Comment by pmario** [comment]
@Jermolene -- https://github.com/TiddlyWiki/TiddlyWiki5/pull/8678#issuecomment-2416757479 we found out that adding `$:/tags/Global` and `$:/tags/Macro` can cause backwards compatibility problems with existing plugins that use `$:/tags/Macro` only. 

If the new core code contains both tags, it happens that 3rd party plugins do not work anymore. 

@BurningTreeC has an idea shown in the OP -- More testing is needed.

**Comment by pmario** [comment]
@Jermolene -- I did test it with `GlobalImportFilter`

```
[[$:/core/ui/PageMacros]] [all[shadows+tiddlers]tag[$:/tags/Global]!is[draft]] [all[shadows+tiddlers]tag[$:/tags/Macro]!is[draft]]
```

and `$:/core/macros/keyboard-driven-input` having both tags. The ASP-plugin works without any changes. 

**So the question is**: Should we change the order in `GlobalImportFilter`

**Comment by BurningTreeC** [comment]
I believe it's a "who was first" question

**Comment by Jermolene** [comment]
Apologies I'm finding the thread hard to follow. Why have we even got tiddlers with both tags?

**Comment by pmario** [comment]
>Apologies I'm finding the thread hard to follow. Why have we even got tiddlers with both tags?

Because of this commit and the suggestion within it. https://github.com/TiddlyWiki/TiddlyWiki5/commit/070327cb5708cfad104e2f325115203b499af631

**Comment by pmario** [comment]
>What is the rationale for the change? I would expect that a tiddler with the global tag would take precedence over one with the deprecated "macro" tag.

Backwards compatibility. Plugins with the older $:/tags/Macro only did expect, that they are executed after the core macros. So they are able to overwrite core macros if needed. 

Now the new GlobalImportFilter changes that rule and breaks existing code **and that's an issue**.

**Comment by Jermolene** [comment]
Could a solution be for the core to continue to use `$:/tags/Macro` for all the core globals?

**Comment by pmario** [comment]
I think Yes. -- But we have users, that will complain, that this term is not right anymore for `\function \procedures and \widgets` ... because they are "variables" now.

So we have to decide, what we want.

**Comment by pmario** [comment]
We could keep the `$:/tags/Macros` for some more time, but remind all plugin authors, that they should use both `$:/tags/Global` and `$:/tags/Macros` if they want to support v5.2.x and down.

Once most plugin authors use `$:/tags/Global` we should be good to switch in the core too.

**Comment by Jermolene** [comment]
Fixed in 0a03ac8219b5472bf8a0292d4ea4d61c09265c4a

**Comment by Jermolene** [comment]
I reverted the recent changes so that now the core global macros only use $:/tags/Macro

**Comment by pmario** [comment]
I'll close the issue
